src/main/typescript/components/Header.tsx

```diff
@@ -83,43 +85,58 @@ interface HeaderProps {
+class Header extends Component<HeaderProps> {
 
-    return <>
-        <NavBar>
-            <NavBarLink to={homeRoute} title="Home">Home</NavBarLink>
-            <NavBarLink to={registerRoute} title="Register to get access to EASY">Register</NavBarLink>
-            {...loginNavBar}
-        </NavBar>
+    componentDidUpdate() {
+        const { isLoggedIn, loginName, getUser } = this.props
+        if (loginName == undefined) {
+            getUser()
+        }
+    }
 
-        <LogosHeaders>
-            <BrandLogo className="col-6 col-md-3 col-lg-2"
-                       id="dans-logo"
-                       src={logo_dans}
-                       alt="DANS - Data Archiving and Networked Services"/>
-            <BrandLogo className="col-6 col-md-2 offset-md-2 col-lg-2 offset-lg-3"
-                       id="easy-logo"
-                       height="25px"
-                       src={logo_easy}
-                       alt="EASY"/>
-        </LogosHeaders>
-        {/* TODO not sure if this <hr/> will stay, but I think it is useful during development */}
-        <hr/>
-    </>
+    render() {
+        const { isLoggedIn, loginName, signout } = this.props
+        const loginNavBar = isLoggedIn
+            ? [
+                <span key="loginName" className="navbar-text">{loginName}</span>,
+                <NavBarLink key="my datasets" to={depositOverviewRoute}>My Datasets</NavBarLink>,
+                <Link onClick={signout} className="nav-link logoff" key="log out" to="/" title="Log out">Log out</Link>,
+            ]
+            : [<NavBarLink key="login" to={loginRoute} title="Login to EASY">Login</NavBarLink>]
+
+        return <>
+            <NavBar>
+                <NavBarLink to={homeRoute} title="Home">Home</NavBarLink>
+                <NavBarLink to={registerRoute} title="Register to get access to EASY">Register</NavBarLink>
+                {...loginNavBar}
+            </NavBar>
+
+            <LogosHeaders>
+                <BrandLogo className="col-6 col-md-3 col-lg-2"
+                           id="dans-logo"
+                           src={logo_dans}
+                           alt="DANS - Data Archiving and Networked Services"/>
+                <BrandLogo className="col-6 col-md-2 offset-md-2 col-lg-2 offset-lg-3"
+                           id="easy-logo"
+                           height="25px"
+                           src={logo_easy}
+                           alt="EASY"/>
+            </LogosHeaders>
+            {/* TODO not sure if this <hr/> will stay, but I think it is useful during development */}
+            <hr/>
+        </>
+    }
 }
 
-const mapStateToProps = (state: AppState) => ({
-    isLoggedIn: state.user.isAuthenticated,
-    loginName: "[displayName]",
-})
+const mapStateToProps = (state: AppState) => {
+    console.log(state)
+    return ({
+        isLoggedIn: state.authenticatedUser.isAuthenticated,
+        loginName: state.user.displayName || undefined,
```

Comment:
this shouldn't even compile, I'd say! `loginName` is not optional, so `undefined` should not even be allowed.
I would suggest to make `loginName` optional anyway, because the `displayName` is optional as well. (`loginName?: string`)

Reply:
mmm, maybe not, depending on whether the `displayName` is in fact optional itself... otherwise remove the `|| undefined`